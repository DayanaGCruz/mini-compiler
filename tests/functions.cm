int add(int lhs, int rhs) {
  int sum;

  sum = lhs + rhs;
  return sum;
}

int fortyTwo() {
  return 42;
}

char echoChar(char input) {
  return input;
}

void main() {
  int total;
  int nested;

  total = add(7, 5);
  print(total);

  nested = add(add(1, 2), fortyTwo());
  print(nested);

  print(echoChar('z'));
}
