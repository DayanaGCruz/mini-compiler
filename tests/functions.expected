12
44
z
